src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -202,13 +209,77 @@ protected override Task CreateVersionsAsync(CancellationToken cancellationToken)
             // Add disabled versions
             AddBlockedVersions(blockedVersions);
 
+            // Update the view and add the filter
+            VersionsView = new CollectionViewSource() { Source = Versions }.View;
+
+            if (IsProjectPackageReference)
+            {
+                VersionsView.Filter += VersionsFilter;
+            }
+
             SelectVersion();
 
             OnPropertyChanged(nameof(Versions));
+            OnPropertyChanged(nameof(VersionsView)); // This can change the selected version if a version is added after SelectedVersion is set
+
+            // If the selected version changed, reset it to the correct one
+            if (_nugetProjects.Any() &&
+                IsProjectPackageReference &&
+                SelectedVersion != FirstDisplayedVersion)
+            {
+                SelectedVersion = FirstDisplayedVersion;
+                UserInput = SelectedVersion?.ToString();
+            }
 
             return Task.CompletedTask;
         }
 
+        private bool IsBeforeNullSeparator { get; set; }
+
+        private bool VersionsFilter(object o)
+        {
+            var version = o as DisplayVersion;
+            // If the text is empty or is the insalled version we should show all the versions like if there where no filtering
+            if (string.IsNullOrEmpty(UserInput) || UserInput.Equals(FirstDisplayedVersion?.ToString(), StringComparison.OrdinalIgnoreCase))
+            {
+                if (IsBeforeNullSeparator) IsBeforeNullSeparator = false;
+                return true;
+            }
+
+            // Handle of *, that this will show all versions other than the installed version, the suggested version and the null separator
+            if (UserInput.Equals("*", StringComparison.OrdinalIgnoreCase))
+            {
+                if (version == null) IsBeforeNullSeparator = true;
+
+                if (!IsBeforeNullSeparator || version == null)
+                {
+                    return false;
+                }
+                return true;
+            }
+            else if (IsBeforeNullSeparator) IsBeforeNullSeparator = false;
```

Comment:
so technically we don't have a [coding guideline](https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md) rule (at least, not one that I saw in a 15 second skim), I think that even though this is a single line expression, since it follows an `if` statement that did have a `{ }` block, I think it will help readability if you add a scoping block.

```suggestion
            else if (IsBeforeNullSeparator)
            {
                IsBeforeNullSeparator = false;
            }
```